Codes/coordinate_initialization.py

```diff
@@ -0,0 +1,228 @@
+	yr=512
+	xt,yt=get_ratio_points(x1,x2,y1,y2,1,1)
+	xd,yd=get_ratio_points(x2,x3,y2,y3,1,1)
+	m=(yd-yt)/(xd-xt)
+	c=yd-(m*xd)
+	xr=(yr-c)/m;
+	#distance in pixels
+	distance = math.sqrt(math.pow(xc-xr,2)+math.pow(yc-yr,2))
+	distance=distance*mm_per_pix
+
+#****************************************
+#Function Name:cal_distance_for_black
+#  * Input: 4 coord. of border black 
+#  * Output: none
+#  * Logic: get center line of black border
+#			cal. the distance from midpoint of photo 
+#  * Example Call: cal_distance_for_black(1,2,3,4,5,6,7,8)
+#****************************************
+def cal_distance_for_black(x1,x2,x3,x4,y1,y2,y3,y4):
+	xc,yc=(640,512)
+	xr=640
+	xt,yt=get_ratio_points(x1,x3,y1,y3,1,1)
+	xd,yd=get_ratio_points(x2,x4,y2,y4,1,1)
+	m=(yd-yt)/(xd-xt)
+	c=yd-(m*xd)
+	yr=m*xr+c
+	#distance in pixels
+	distance = math.sqrt(math.pow(xc-xr,2)+math.pow(yc-yr,2))
+	distance=distance*mm_per_pix
+
+#****************************************
+#Function Name:go_for_y
+#  * Input: int final-whether to go for final run
+#  * Output: none
+#  * Logic: will generate the serial string
+# 			and communicate to arduino, followed
+#			by confirmation  (for y direction)
+#  * Example Call:go_for_y()
+#****************************************
+def go_for_y(final):
+	# communication string : @ + x_coord + y_coord + z_coord + seed_bit + water_bit + initialization_bit + !
+	arduino_confirm=False
+	mystring=""
+	if final == 0 :
+		comm_string="@"+"0000"+str(dist_y_bord).zfill(3)+"000"+"002"+"!"
+	elif final == 1 :
+		curr_y=distance
+	ser.write(comm_string.encode("ascii"))
+	while arduino_confirm == False :
+		line_in=ser.readline()
+		for char in line_in:
+			mystring = mystring + chr(char)
+		if mystring == "@done!" :
+			arduino_confirm=True
+
+#****************************************
+#Function Name:go_for_x
+#  * Input: int final-whether to go for final run
+#  * Output: none
+#  * Logic: will generate the serial string
+# 			and communicate to arduino, followed
+#			by confirmation(for x direction)  
+#  * Example Call: go_for_x()
+#****************************************
+
+def go_for_x(final):
+	arduino_confirm=False
+	mystring=""
+	if final == 0 :
+		comm_string="@"+str(dist_x_bord).zfill(4)+"000"+"000"+"002"+"!"
+	elif final == 1 :
+		curr_x=distance
+		comm_string="@"+str(curr_x).zfill(4)+str(curr_y).zfill(3)+"000"+"001"+"!"
+	ser.write(comm_string.encode("ascii"))
+	while arduino_confirm == False :
+		line_in=ser.readline()
+		for char in line_in:
+			mystring = mystring + chr(char)
+		if mystring == "@done!" :
+			arduino_confirm=True
+
+#****************************************
+#Function Name:image_process_for_coord
+#  * Input: none
+#  * Output: none 
+#  * Logic: will take photo and process it 
+#			to look for red border.
+#			if found will call go_for_y
+#			after changing dist_y_bord
+#			as calculated
+#  * Example Call: image_process_for_coord()
+#****************************************
+
+def image_process_for_coord():
```

Comment:
can you please add the explanation of your algorithm?
1. add explnation here, or
2. add it in documentation and mention that here.
In explanation, write about purpose of every function that you are writing as you have already for most of the functions.